changelog.d/issue-4683

```diff
@@ -0,0 +1,4 @@
+synopsis: 'cabal check' to fail when no upper bounds for base or Cabal are present in setup dependencies
```

Comment:
Does `check` fail (error out) or does it warn (emit warning)?

Reply:
It is a fatal failure both for component dependencies (for some time now) and setup dependencies (as implemented in this PR). That's, yet again, for base (and Cabal for setup dependencies) only.

Reply:
OK, got it.